src/olympia/blocklist/tests/test_cron.py

```diff
@@ -538,49 +525,215 @@ def test_upload_stash_unless_enough_changes(self):
-                create_stash=False,
-            )
-        ]
-        mlbf = MLBF.load_from_storage(self.current_time)
-        assert mlbf.storage.exists(mlbf.filter_path(BlockType.BLOCKED))
-        assert not mlbf.storage.exists(mlbf.stash_path)
 
-        with override_switch('enable-soft-blocking', active=True):
-            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
-            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
-            upload_mlbf_to_remote_settings()
-            self.mocks['olympia.blocklist.cron.upload_filter.delay'].assert_called_with(
-                self.current_time,
-                filter_list=[BlockType.BLOCKED.name],
-                create_stash=False,
-            )
-            mlbf = MLBF.load_from_storage(self.current_time)
-            assert mlbf.storage.exists(mlbf.filter_path(BlockType.BLOCKED))
-            assert not mlbf.storage.exists(mlbf.stash_path)
+        self._test_upload_stash_and_filter(
+            # Soft blocking is enabled, so we expect a stash and a filter
+            enable_soft_blocking=True,
+            expected_stash={
+                'blocked': 1,
+                # Expect no soft blocks becuase there are enough for a filter
+                'softblocked': 0,
+                'unblocked': 0,
+            },
+            # Expect a soft blocked filter
+            expected_filters=[BlockType.SOFT_BLOCKED],
+        )
+
+    def test_upload_soft_blocked_stash_and_blocked_filter(self):
+        # Enough soft blocks for a stash
+        self._block_version(is_signed=True, block_type=BlockType.SOFT_BLOCKED)
+        # Enough blocked versions for a filter
+        self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+        self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+
+        self._test_upload_stash_and_filter(
+            enable_soft_blocking=False,
+            # Expect no stash because there are enough blocked versions for a filter
+            # and soft blocking is disabled
+            expected_stash=None,
+            # Expect a blocked filter
+            expected_filters=[BlockType.BLOCKED],
+        )
+
+        # Now try again with soft blocking enabled
+        self._test_upload_stash_and_filter(
+            enable_soft_blocking=True,
+            # Expect a stash and a blocked filter
+            expected_stash={
+                # Expect no blocked versions because there are enough for a filter
+                'blocked': 0,
+                # Expect a soft blocked version when there is one
+                # and soft blocking is enabled
+                'softblocked': 1,
+                # There are no unblocked versions
+                'unblocked': 0,
+            },
+            # Expect a blocked filter
+            expected_filters=[BlockType.BLOCKED],
+        )
+
+    def test_upload_soft_blocked_stash_and_blocked_filter_2(self):
```

Comment:
Please find a better name :D

Reply:
How about `test_upload_soft-blocked_stash_and_blocked_filter_deux`? 🙃

Reply:
This test is a dupe of `test_upload_blocked_stash_and_softblock_filter` just removing it.